apps/passport-client/src/validateState.ts

```diff
@@ -0,0 +1,174 @@
+import { User, requestLogToServer } from "@pcd/passport-interface";
+import { PCDCollection } from "@pcd/pcd-collection";
+import {
+  SemaphoreIdentityPCD,
+  SemaphoreIdentityPCDPackage
+} from "@pcd/semaphore-identity-pcd";
+import { appConfig } from "./appConfig";
+import { AppState } from "./state";
+
+/**
+ * Determines whether the app's global state as represented by {@link AppState} object
+ * contains valid data. If it does not, returns the set of things that are incorrect about
+ * it in an array of human interpretable strings. If there are no errors, returns an
+ * empty array.
+ */
+export function validateState(state: AppState): string[] {
+  const validationErrors: string[] = [];
+
+  if (!state.self) {
+    validationErrors.push("missing 'self' field from app state");
+  }
+
+  if (!state.identity) {
+    validationErrors.push("missing 'identity' field from app state");
+  }
+
+  if (!state.encryptionKey) {
+    validationErrors.push("missing 'encryption' field from app state key");
+  }
+
+  if (!state.pcds) {
+    validationErrors.push("missing 'pcds' field from app state");
+  }
+
+  if (state.pcds.size() === 0) {
+    validationErrors.push("'pcds' field in app state contains no pcds");
+  }
+
+  const identityPCDFromCollection = state.pcds.getPCDsByType(
```

Comment:
I think there's an opportunity here to change the lookup algorithm (in all places) to take `identity` as an input rather than relying on order.  Not strictly necessary for this PR, though.

Reply:
I think I'd prefer to leave that out of this PR.